samples/server/petstore/nancyfx/src/IO.Swagger/Modules/UserModule.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using Nancy;
+using Nancy.ModelBinding;
+using System.Collections.Generic;
+using Sharpility.Base;
+using IO.Swagger.v2.Models;
+using IO.Swagger.v2.Utils;
+using NodaTime;
+
+namespace IO.Swagger.v2.Modules
+{ 
+
+    /// <summary>
+    /// Module processing requests of User domain.
+    /// </summary>
+    public sealed class UserModule : NancyModule
+    {
+        /// <summary>
+        /// Sets up HTTP methods mappings.
+        /// </summary>
+        /// <param name="service">Service handling requests</param>
+        public UserModule(UserService service) : base("/v2")
+        { 
+            Post["/user"] = parameters =>
+            {
+                var body = this.Bind<User>();
+                service.CreateUser(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Post["/user/createWithArray"] = parameters =>
+            {
+                var body = this.Bind<List<User>>();
+                service.CreateUsersWithArrayInput(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Post["/user/createWithList"] = parameters =>
+            {
+                var body = this.Bind<List<User>>();
+                service.CreateUsersWithListInput(Context, body);
+                return new Response { ContentType = "application/json"};
+            };
+
+            Delete["/user/{username}"] = parameters =>
+            {
+                var username = Parameters.ValueOf<string>(parameters, Context.Request, "username", ParameterType.Path);
```

Comment:
@jchannon thanks for reviewing the code (and sorry for late reply as I missed your question)

I think it's trying to obtain `username` from the path. e.g. if path is `/user/wing328`, then `username` will store "wing328". In other words, it's not a query string.